File: sdks/java/src/main/java/dev/atap/sdk/model/DIDCommMessage.java
package dev.atap.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * A DIDComm message from the inbox.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class DIDCommMessage {

    @JsonProperty("id")
    private String id;

    @JsonProperty("sender_did")
    private String senderDid;

    @JsonProperty("message_type")
    private String messageType;

    @JsonProperty("payload")
    private String payload;

    @JsonProperty("created_at")
    private String createdAt;

    public DIDCommMessage() {
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getSenderDid() { return senderDid; }
    public void setSenderDid(String senderDid) { this.senderDid = senderDid; }

    public String getMessageType() { return messageType; }
    public void setMessageType(String messageType) { this.messageType = messageType; }

    public String getPayload() { return payload; }
    public void setPayload(String payload) { this.payload = payload; }

    public String getCreatedAt() { return createdAt; }
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }
}

File: sdks/php/src/Model/DIDCommMessage.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Model;

/**
 * A DIDComm message from the inbox.
 */
final class DIDCommMessage
{
    public function __construct(
        public readonly string $id,
        public readonly string $senderDid = '',
        public readonly string $messageType = '',
        public readonly string $payload = '',
        public readonly string $createdAt = '',
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            senderDid: (string) ($data['sender_did'] ?? ''),
            messageType: (string) ($data['message_type'] ?? ''),
            payload: (string) ($data['payload'] ?? ''),
            createdAt: (string) ($data['created_at'] ?? ''),
        );
    }
}
